datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -921,6 +938,70 @@ public void waitForDependencies(DatastreamTaskImpl task, Duration timeout) {
     });
   }
 
+  /**
+   * Clean up partition movement info for a particular datastream
+   * @param connectorType Connector name
+   * @param datastreamGroupName the name of datastream group
+   */
+  public void cleanUpPartitionMovement(String connectorType, String datastreamGroupName) {
+    String path = KeyBuilder.getTargetAssignment(_cluster, connectorType, datastreamGroupName);
+    if (_zkclient.exists(path)) {
+      _zkclient.deleteRecursive(path);
+    }
+    LOG.info("clean up Target assignment info for {}", datastreamGroupName);
+  }
+
+  /**
+   * Get the datastream group which contains the partition movement info
+   * @param connectorType Connector name
+   */
+  public List<String> getDatastreamsNeedPartitionMovement(String connectorType) {
+    String path = KeyBuilder.getTargetAssignmentBase(_cluster, connectorType);
+    if (_zkclient.exists(path)) {
+      return _zkclient.getChildren(path);
+    }
+    return Collections.emptyList();
+  }
+
+  /**
+   * Get a partition movement info for a particular datastream
+   * @param connectorType Connector name
+   * @param datastreamGroupName the name of datastream group
+   * @return The target assignment mapping information with the instance name -> topicPartition
+   */
+  public Map<String, Set<String>> getPartitionMovement(String connectorType, String datastreamGroupName) {
+    Map<String, Set<String>> result = new HashMap<>();
+
+    String path = KeyBuilder.getTargetAssignment(_cluster, connectorType, datastreamGroupName);
+    if (_zkclient.exists(path)) {
+      List<String> nodes = _zkclient.getChildren(path);
+      Collections.sort(nodes);
+      for (String node : nodes) {
+        String content = _zkclient.readData(path + '/' + node);
+        TargetAssignment assignment = TargetAssignment.fromJson(content);
+
+        //Compute the correct instance
+        Map<String, String> hostInstanceMap = new HashMap<>();
+        if (_liveInstancesProvider.getLiveInstances() != null) {
+          _liveInstancesProvider.getLiveInstances().forEach(instance -> {
+            try {
+              hostInstanceMap.put(instance.substring(0, instance.lastIndexOf('-')), instance);
+            } catch (Exception ex) {
+              LOG.info("Fail to parse instance name {}", instance, ex);
+            }
+          });
+        }
+        if (hostInstanceMap.containsKey(assignment.getTargetHost()) && assignment.getPartitionNames() != null) {
+          LOG.info("Added assignment {} {}", assignment.getTargetHost(), assignment.getPartitionNames());
+          result.put(hostInstanceMap.get(assignment.getTargetHost()), new HashSet<>(assignment.getPartitionNames()));
+        } else {
+          LOG.warn("instance {} not found in the map {}", assignment.getTargetHost(), hostInstanceMap.keySet());
```

Comment:
The log statement should be more clear on where the instance is not found. Map is a datastructure not the location that people reading the log can understand.

Reply:
Fixed, thanks for suggestion.